lib/Jsrt/Core/JsrtCore.cpp

```diff
@@ -77,7 +77,18 @@ JsParseModuleSource(
         SourceContextInfo* sourceContextInfo = scriptContext->GetSourceContextInfo(sourceContext, nullptr);
         if (sourceContextInfo == nullptr)
         {
-            sourceContextInfo = scriptContext->CreateSourceContextInfo(sourceContext, nullptr, 0, nullptr, nullptr, 0);
+            if(moduleRecord->GetModuleURL() != nullptr)
+            {
+                Js::JavascriptString *moduleURL = Js::JavascriptString::FromVar(moduleRecord->GetModuleURL());
```

Comment:
I think I would prefer if this block read as:

```c++
const char16 *moduleURLSz = nullptr;
size_t moduleURLLen = 0;
if (moduleRecord->GetModuleURL())
{
  Js::JavascriptString *moduleURL = Js::JavascriptString::FromVar(moduleRecord->GetModuleURL());
  moduleURLSz = moduleURL->GetSz();
  moduleURL->GetLength();
}
sourceContextInfo = scriptContext->CreateSourceContextInfo(sourceContext, moduleURLSz, moduleURLLen, nullptr, nullptr, 0);
```

Maybe just a personal preference, but this is simpler for me to understand.